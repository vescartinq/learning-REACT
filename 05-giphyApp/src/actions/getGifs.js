export const getGifs = async (category) => {
  const apiKey = 'HPqG9gDBAf3Yu78aQYSjVQp1XvmOPRCV';

  const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(
    category
  )}&limit=10&api_key=${apiKey}`;

  const resp = await fetch(url);
  const { data } = await resp.json();

  const gifs = data.map((img) => {
    return {
      id: img.id,
      title: img.title,
      url: img.images?.downsized_medium.url,
    };
  });

  return gifs;
};
