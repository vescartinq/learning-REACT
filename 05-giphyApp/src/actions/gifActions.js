export const getGifs = async (category) => {
  const apiUrl = 'api.giphy.com/v1/gifs';
  const apiKey = 'HPqG9gDBAf3Yu78aQYSjVQp1XvmOPRCV';
  const limitGifs = 20;

  // encodeURI elimina espacios del input por + para la url
  const url = `https://${apiUrl}/search?q=${encodeURI(
    category
  )}&limit=${limitGifs}&api_key=${apiKey}`;

  const resp = await fetch(url);
  const { data } = await resp.json();

  const gifs = data.map((img) => {
    return {
      id: img.id,
      title: img.title,
      url: img.images?.downsized_medium.url,
    };
  });

  return gifs;
};
