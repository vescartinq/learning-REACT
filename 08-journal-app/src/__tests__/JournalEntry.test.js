import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { activeNote } from '../actions/notes';
import { JournalEntry } from '../components/journal/JournalEntry';

// jest.mock('../actions/notes', () => ({
//   activeNote: jest.fn(),
// }));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
  id: 10,
  date: 0,
  title: 'Hola',
  body: 'Mundo',
  url: 'https://algunafoto.com/foto.jpg',
};

const wrapper = mount(
  <Provider store={store}>
    <JournalEntry {...note} />
  </Provider>
);

describe('JournalEntry', () => {
  test('should render <JournalEntry/> component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should activate note', () => {
    wrapper.find('.journal__entry').prop('onClick')();

    expect(store.dispatch).toHaveBeenCalledWith(
      activeNote(note.id, { ...note })
    );
  });
});
