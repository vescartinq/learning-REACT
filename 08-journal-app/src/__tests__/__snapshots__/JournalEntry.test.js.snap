// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalEntry should render <JournalEntry/> component 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://algunafoto.com/foto.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
       
      Thursday
       
    </span>
    <h4>
       
      1st
       
    </h4>
  </div>
</div>
`;
